### DISCUSSION TITLE
How to get possible values of a stack var?

### DISCUSSION THREAD

**Discussion by d0mnik** [Vector35/binaryninja-api#4285] [API] [answered]
According to the api documentation, the `get_possible_values` is only for IL instructions. Is there a way to accomplish the same for stack vars?
I'm looking to obtain possible values of a stack var at a given point in the basic block(likely just a few instruction after a `mov {var_4}, 0x10` for example).

**Comment by CouleeApps** [comment, marked as answer]
You can use `LowLevelILInstruction.get_possible_stack_contents_after` / `MediumLevelILInstruction.get_possible_stack_contents_after` to find possible stack values after an LLIL/MLIL instruction. You can use `Function.get_low_level_il_here` to look up an LLIL instruction at whatever address you want to inspect.


```
>>> current_function.get_low_level_il_at(here)
<llil: [ebp - 4 {var_8_3}].d = 1>
>>> current_function.get_low_level_il_at(here).get_possible_stack_contents_after(-8, 4)
<const 0x1>
```
